latextools_utils/tex_directives.py

```diff
@@ -0,0 +1,116 @@
+from __future__ import print_function
+
+import codecs
+import re
+import sublime
+import sys
+import traceback
+
+if sys.version_info < (3, 0):
+    strbase = basestring
+else:
+    strbase = str
+
+TEX_DIRECTIVE = re.compile(
+    r'%+\s*!(?:T|t)(?:E|e)(?:X|x)\s+([\w-]+)\s*=\s*' +
```

Comment:
Re 1. It's implemented this way for consistency with [Texworks](https://github.com/TeXworks/texworks/blob/master/src/TeXDocument.cpp#L3033) and TeXshop, which understand `%` as a comment and `!TEX` as the actual directive.

Re 2: Yes. This is an artefact of trying to use the same re both here and [in the Atom version](https://github.com/msiniscalchi/atom-latextools/blob/master/lib/parsers/tex-directive-parser.coffee#L3). I suppose I could've used flags in both places.